File: LexicographicalNumbers.py
class Solution:
    def lexicalOrder(self, n):
        """
        :type n: int
        :rtype: List[int]
        """
        def helper(v, n):
            if v <= n:
                result.append(v)
            else:
                return
            helper(v * 10, n)
            if v % 10 < 9:
                helper(v + 1, n)
        result = []
        helper(1, n)
        return result


if __name__ == '__main__':
    n = int(input())
    solution = Solution()
    print(solution.lexicalOrder(n))

File: LexicographicalNumbers.cpp
#include<iostream>
#include<vector>

using namespace std;

class Solution {
public:
    vector<int> lexicalOrder(int n) {
        vector<int> result;
    	
    	helper(result, 1, n);
    	
    	return result;
    }
    
    void helper(vector<int>& result, int v, int n) {
    	if(v <= n) result.push_back(v);
    	else return;
    	helper(result, v * 10, n);
    	
    	if(++v % 10 != 0) helper(result, v, n);
	}
};

int main() {
	
	int n;
	cin>>n;
	
	Solution solution = Solution();
	vector<int> result = solution.lexicalOrder(n);
	
	for(int i = 0; i < result.size(); i++) cout<<result[i]<<"  ";
	
	return 0;
	
}
